pravega-client-examples/README.md

```diff
@@ -0,0 +1,132 @@
+
+```
+$ bin/helloWorldReader [-scope myScope] [-name myStream] [-uri tcp://127.0.0.1:9090]
+```
+
+All args are optional, if not included, the defaults are:
+
+ * scope - "examples"
+ * name - "helloStream"
+ * uri - "tcp://127.0.0.1" (the URI to one of the controller nodes
+
+The program reads all the events from the stream with given scope/stream name and prints each event to the console.
+
+
+## Console Reader and Writer Example
+This example includes two applications, a ConsoleReader and a ConsoleWriter.  You might want to run ConsoleReader in one window and ConsoleWriter in another window.
+
+### ConsoleReader
+Use this application to launch an application that reads from a stream and emits all of the Events onto the console.  This application runs until you terminate it.
+
+```
+$ bin/consoleReader [-scope myScope] [-name myStream] [-uri tcp://127.0.0.1:9090]
+```
+
+All args are optional, if not included, the defaults are:
+
+ * scope - "examples"
+ * name - "someStream"
+ * uri - "tcp://127.0.0.1" (the URI to one of the controller nodes
+
+### ConsoleWriter
+Use this application to write to streams or transactions, and manage transactions.
+
+The application uses the console to present an interactive DSL environment that presents operations to write events to
+a stream or into a transaction.  In addition, it presents operations to begin, commit, abort, ping, check status on and
+retrieve the id of a transaction.
+
+```
+$ bin/consoleWriter [-scope myScope] [-name myStream] [-uri tcp://127.0.0.1:9090]
+```
+
+All args are optional, if not included, the defaults are:
+
+ * scope - "examples"
+ * name - "someStream"
+ * uri - "tcp://127.0.0.1" (the URI to one of the controller nodes
+
+## State Synchronizer
+This example illustrates the use of the Pravega StateSynchronizer.
+
+The application implements a SharedMap object using StateSynchronizer.  We implement a SharedConfig object using
+the SharedMap.  The SharedConfig simulates the idea of a properties configuration that needs to be kept in sync
+across multiple processes.
+
+To demonstrate manipulating the properties of the SharedConfig object, we provide a CLI.
+
+```
+$ bin/sharedConfigCli [-scope myScope] [-name myStream] [-uri tcp://127.0.0.1:9090]
+```
+
+Use the simple DSL to GET, PUT, REMOVE keys from the SharedConfig object identified by scope and name.
+
+It is worthwhile to launch two or more separate CLIs in separate windows using the same settings and observe how changes in one
+CLI process are not visible in another CLI process until that other CLI process invokes REFRESH.
+
+## NoopReader
+
+An example of a simple reader that continually reads the contents of any stream. A binary serializer is used so it works against any event types. The sample emits basic information about number of events/bytes read every 30 seconds. 
+
+```
+$ bin/noopReader [--uri tcp://127.0.0.1:9090] [--stream <SCOPE>/<STREAM>]
+```
```

Comment:
Can we include stream cuts also? if there is way to execute them.
Also, it will be great if we can follow the order mentioned in the https://github.com/pravega/pravega-samples/wiki/Repository-Structure-Reorganization#repository-structure-proposal

Reply:
Actually I realized that the README version was not the correct one (which has the `streamcuts` sample execution instructions in the same order that the main table). Thanks.